bundles/org.openhab.binding.gce/src/main/resources/ESH-INF/thing/channels.xml

```diff
@@ -0,0 +1,212 @@
+			<channel id="4" typeId="counter">
+				<label>Counter 4</label></channel>
+			<channel id="4-duration" typeId="duration" />
+			<channel id="5" typeId="counter">
+				<label>Counter 5</label></channel>
+			<channel id="5-duration" typeId="duration" />
+			<channel id="6" typeId="counter">
+				<label>Counter 6</label></channel>
+			<channel id="6-duration" typeId="duration" />
+			<channel id="7" typeId="counter">
+				<label>Counter 7</label></channel>
+			<channel id="7-duration" typeId="duration" />
+			<channel id="8" typeId="counter">
+				<label>Counter 8</label></channel>
+			<channel id="8-duration" typeId="duration" />
+		</channels>
+	</channel-group-type>
+
+	<channel-group-type id="A">
+		<label>Analog Inputs</label>
+		<channels>
+			<channel id="1" typeId="analogInput" >
+				<label>Analog Input 1</label></channel>
+			<channel id="1-duration" typeId="duration" />
+			<channel id="2" typeId="analogInput" >
+				<label>Analog Input 2</label></channel>
+			<channel id="2-duration" typeId="duration" />
+			<channel id="3" typeId="analogInput" >
+				<label>Analog Input 3</label></channel>
+			<channel id="3-duration" typeId="duration" />
+			<channel id="4" typeId="analogInput" >
+				<label>Analog Input 4</label></channel>
+			<channel id="4-duration" typeId="duration" />
+		</channels>
+	</channel-group-type>
+
+	<channel-type id="digitalInput">
+		<item-type>Contact</item-type>
+		<label>Digital Input</label>
+		<category>Contact</category>
+		<state pattern="%s" readOnly="true"></state>
+		<config-description>
+			<parameter name="debouncePeriod" type="integer" min="0" max="3000" unit="ms">
+				<label>Debounce Time</label>
+				<default>0</default>
+			</parameter>
+			<parameter name="longPressTime" type="integer" step="1000" min="0" max="5000" unit="ms">
+				<label>Long Press Time</label>
+				<description>Long press time in milliseconds.</description>
+				<default>0</default>
+			</parameter>
+			<parameter name="pulsePeriod" type="integer" step="500" min="0" max="50000" unit="ms">
+				<label>Pulse Period</label>
+				<description>Pulse period in milliseconds.</description>
+				<default>0</default>
+			</parameter>
+			<parameter name="pulseTimeout" type="integer" step="500" min="0" max="50000" unit="ms">
+				<label>Pulse Timeout</label>
+				<description>Maximum period for sending pulses in milliseconds.</description>
+				<default>0</default>
+			</parameter>
+		</config-description>
+	</channel-type>
+
+	<channel-type id="digitalOutput">
+		<item-type>Switch</item-type>
+		<label>Digital Output</label>
+		<config-description>
+			<parameter name="pulse" type="boolean">
+				<label>Pulse Output</label>
+				<default>false</default>
+			</parameter>
+		</config-description>
+	</channel-type>
+
+	<channel-type id="counter">
+		<item-type>Number</item-type>
+		<label>Counter</label>
+		<config-description>
+			<parameter name="pullFrequency" type="integer" required="false" min="500" max="60000" unit="ms">
```

Comment:
Actually it's not a frequency (Hz), but an interval. Sorry, I had to comment on this :)